pkg/controller/ssp/metrics/reconcile.go

```diff
@@ -0,0 +1,66 @@
+package metrics
+
+import (
+	"fmt"
+
+	promv1 "github.com/coreos/prometheus-operator/pkg/apis/monitoring/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/intstr"
+
+	"kubevirt.io/ssp-operator/pkg/controller/ssp/common"
+)
+
+func WatchTypes() []runtime.Object {
+	return []runtime.Object{&promv1.PrometheusRule{}}
+}
+
+func Reconcile(request *common.Request) error {
+	rule := newPrometheusRule(request.Namespace)
+
+	err := request.SetControllerReferenceFor(rule)
+	if err != nil {
+		return err
+	}
+
+	var found promv1.PrometheusRule
+	err = request.Client.Get(request.Context,
+		types.NamespacedName{Name: rule.Name, Namespace: rule.Namespace},
+		&found)
+	if errors.IsNotFound(err) {
+		request.Logger.Info(fmt.Sprintf("Creating PrometheusRule: %s", rule.Name))
+		return request.Client.Create(request.Context, rule)
+	}
+	if err != nil {
+		return err
+	}
+
+	found.Spec = rule.Spec
+	request.Logger.Info(fmt.Sprintf("Updating PrometheusRule: %s", found.Name))
+	return request.Client.Update(request.Context, &found)
```

Comment:
From my experience, `Update()`ing resources this way might cause an infinite reconciliation loop (each update triggers another reconciliation loop which triggers another update and so on, even if the updated resource was not actually changed), have you encountered something to that effect?
If so, you can use `Patch()`, which does not trigger a reconcile loop if the patched resource was not actually changed.

Reply:
In my test it did not cause an infinite loop, but the update was executed twice. I will check it again.

Reply:
I have added a check for equality.